### ISSUE TITLE
Regarding the warmup in the snippet.

### ISSUE THREAD

**Issue by shreejalt** [TorchSSL/TorchSSL#51]
Hi, 
Thanks a lot for your contribution.

I wanted to ask about the warmup thing. In your flexmatch training code there is args.warmup inside the if condition `if max(pseudo_counter.values()) < len(self.ulb_dset):`

Will it not everytime go in this if condition and `args.warmup` if condition? Because I printed the if statement and through the training it goes in `args.warmup` if condition.  

So ACC to paper the warmup should get over after some time?
Is there a issue in a code or am I understanding it in a wrong way?

Thanks a lot, 
Shreejal

**Comment by Hhhhhhao** [comment]
Can you put your stdout here? So that we can help you debug the problem.
